File: back/schemas/book_copy.py
from pydantic import BaseModel


class BookCopyBase(BaseModel):
    book_id: int
    library_id: int
    inventory_number: str
    status: str  # available, on_loan, maintenance, lost


class BookCopyCreate(BookCopyBase):
    pass


class BookCopyUpdate(BaseModel):
    book_id: int | None = None
    library_id: int | None = None
    inventory_number: str | None = None
    status: str | None = None


class BookCopyResponse(BookCopyBase):
    id: int

    model_config = {"from_attributes": True}


class BookCopyWithDetails(BookCopyResponse):
    book_title: str | None = None
    library_name: str | None = None

File: LibraSmartWPF/Models/BookCopy.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace LibraSmartWPF.Models;

/// <summary>
/// Экземпляры книг (физические копии в библиотеках)
/// </summary>
[Table("book_copies")]
public class BookCopy
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Required]
    [Column("book_id")]
    public int BookId { get; set; }

    [Required]
    [Column("library_id")]
    public int LibraryId { get; set; }

    [Required]
    [MaxLength(50)]
    [Column("inventory_number")]
    public string InventoryNumber { get; set; } = string.Empty;

    [Required]
    [MaxLength(20)]
    [Column("status")]
    public string Status { get; set; } = "available"; // available, on_loan, maintenance, lost

    // Связи
    [ForeignKey(nameof(BookId))]
    public virtual Book Book { get; set; } = null!;

    [ForeignKey(nameof(LibraryId))]
    public virtual Library Library { get; set; } = null!;

    public virtual ICollection<Loan> Loans { get; set; } = new List<Loan>();
}
